conans/requirements_dev.txt

```diff
@@ -1,6 +1,7 @@
 pytest>=6.1.1, <7.0.0; python_version > '3.0'
 pytest>=4.6.11; python_version < '3.0'
 pytest-xdist  # To launch in N cores with pytest -n
+pyfakefs>=4.5.6
```

Comment:
I'd say definitely not worth the complexities and problems for using more third parties, for this very specific use case.

Reply:
Sorry, I didn't realize it was ``requirements_dev.txt``. No problem with this one, if it really helps for the test (if there is an easy way to do the test without it, better, of course)

Reply:
Yup, just used for testing!

I exposes an in-memory filesystem to tests such that python functions that operate on the filesystem read from this instead. 

For the purposes of this test, writing a temp file on the actual file system and reading back is also an option. I wanted to a be a little more explicit making sure that paths like `file:///C:/path/to/file.txt` are resolved properly on Windows - but I think if we used the existing `temp_folder()` functionality in the test, the paths would be implicitly platform-appropriate anyway, so we would achieve the same testing. Let me know if this is preferred!

Reply:
I'd say yes, we already do a ton of ``temp_folder()`` in our tests suite, and I think this could perfectly work, please give it a try.

Reply:
Done! Still does the job :D